net/migrations/0002_alter_connection_internet_exchange_point.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 3.2.4 on 2021-06-24 12:16
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+from net.models import Connection
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("peering", "0075_auto_20210604_1709"),
+        ("net", "0001_initial"),
+    ]
+
+    def deleted_orphaned_connections(apps, schema_editor):
+        connections = Connection.objects.all()
+        for connection in connections:
+            if not connection.internet_exchange_point:
+                print(
+                    f"Deleting Orphaned Connection: {connection} - {connection.ipv4_address}/{connection.ipv6_address}"
+                )
+                connection.delete()
```

Comment:
Couple of remarks here:

1. We should avoid importing models directly in a migration file
2. It's not optimal to proceed to delete like this I would go for a function like the following

```python
    def deleted_orphaned_connections(apps, schema_editor):
        Connection = apps.get_model("net.Connection")
        Connection.objects.filter(internet_exchange_point__isnull=True).delete()
```

Reply:
I think its best we provide some debug to show what is actually being deleted so I still think iterating & deleting is the best option, albeit with not importing the object directly and using a filter.

Reply:
We could catch the number of deleted connections and output it:
```python
deleted, _ = Connection.objects.filter(internet_exchange_point__isnull=True).delete()
if deleted > 0:
    print(f"Deleted {deleted} connections")
```

Not gonna lie, I'm not a fan of iterations when it comes to database queries.

Reply:
Well, we could always do something like this, reducing the amount of DB calls:

```python
connections = Connection.objects.filter(internet_exchange_point__isnull=True)

for connection in connections:
    print(f"Deleting Orphaned Connection: {connection} - {connection.ipv4_address}/{connection.ipv6_address}")
    
Connection.objects.filter(internet_exchange_point__isnull=True).delete()
```